org.eclipse.transformer/src/main/java/org/eclipse/transformer/action/impl/ContainerActionImpl.java

```diff
@@ -201,10 +202,11 @@ protected void apply(String inputPath, ZipInputStream zipInputStream, ZipOutputS
 					// TODO: Should more of the entry details be transferred?
 
 					ZipEntry outputEntry = new ZipEntry(inputName);
-					zipOutputStream.putNextEntry(outputEntry); // throws
-																// IOException
-					FileUtils.transfer(zipInputStream, zipOutputStream, buffer); // throws
-																					// IOException
+					if (putNextEntry(processedClassNames, zipOutputStream, outputEntry)) { // throws
+						// IOException
+						FileUtils.transfer(zipInputStream, zipOutputStream, buffer); // throws
+						// IOException
+					}
 					zipOutputStream.closeEntry(); // throws IOException
```

Comment:
I would also think we must only close the entry _IF_ we actually put the entry in the zip output stream. So the close needs to be hoisted into the if. (Shows need for tests.)

Reply:
good point, my test case likely has the duplicated class last, so I got lucky to not hit this bug.